### PR TITLE
update device init

### PR DESCRIPTION
**Pull request by cl2017**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
This PR refactored the panel init logic
- Updated `DevInit` function at `device.go`
- Change `DevPanel` interface
- `main.go` component startup order optimize
- Remove `util/parse/parse.go` file and function, move to `device.go`

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by cl2017** [comment]
PTAL, thanks~  @fisherxu @WillardHu @wbc6080

File: staging/src/github.com/kubeedge/mapper-framework/_template/mapper/device/device.go [hunk 1]
```diff
@@ -268,25 +270,35 @@ func setVisitor(visitorConfig *driver.VisitorConfig, twin *common.Twin, dev *dri
...
-	//		return err
-	//	}
-	case common.DevInitModeRegister:
-		if err := parse.ParseByUsingRegister(devs, d.models, d.protocols); err != nil {
+func (d *DevPanel) DevInit(deviceList []*dmiapi.Device, deviceModelList []*dmiapi.DeviceModel) error {
+	if len(deviceList) == 0 || len(deviceModelList) == 0 {
+		return ErrEmptyData
+	}
+
+	for _, model := range deviceModelList {
```
Comment (WillardHu) [inline staging/src/github.com/kubeedge/mapper-framework/_template/mapper/device/device.go:278]: Foreach pointer slice with golang version <1.20 can be problematic, you can use:
```golang
for i := range deviceModelList {
    model := deviceModelList[i]
    ...
}
```

File: staging/src/github.com/kubeedge/mapper-framework/_template/mapper/device/device.go [hunk 2]
```diff
@@ -268,25 +270,35 @@ func setVisitor(visitorConfig *driver.VisitorConfig, twin *common.Twin, dev *dri
...
+		cur := parse.ParseDeviceModelFromGrpc(model)
+		d.models[model.Name] = cur
+	}
+
+	for _, device := range deviceList {
```
Comment (WillardHu) [inline staging/src/github.com/kubeedge/mapper-framework/_template/mapper/device/device.go:283]: Foreach pointer slice with golang version <1.20 can be problematic, you can use:
```golang
for i := range deviceModelList {
    device := deviceList[i]
    ...
}
```

File: staging/src/github.com/kubeedge/mapper-framework/_template/mapper/device/device.go [hunk 3]
```diff
@@ -268,25 +270,35 @@ func setVisitor(visitorConfig *driver.VisitorConfig, twin *common.Twin, dev *dri
...
+		commonModel := d.models[device.Spec.DeviceModelReference]
+		protocol, err := parse.BuildProtocolFromGrpc(device)
+		if err != nil {
 			return err
```
Comment (WillardHu) [inline staging/src/github.com/kubeedge/mapper-framework/_template/mapper/device/device.go:287]: Is there a clear enough error message here? If not clear enough, you can supplement the information

File: staging/src/github.com/kubeedge/mapper-framework/_template/mapper/device/device.go [hunk 4]
```diff
@@ -268,25 +270,35 @@ func setVisitor(visitorConfig *driver.VisitorConfig, twin *common.Twin, dev *dri
...
 		}
-	}
+		instance, err := parse.ParseDeviceFromGrpc(device, &commonModel)
+		if err != nil {
+			return err
```
Comment (WillardHu) [inline staging/src/github.com/kubeedge/mapper-framework/_template/mapper/device/device.go:291]: Is there a clear enough error message here? If not clear enough, you can supplement the information

File: staging/src/github.com/kubeedge/mapper-framework/_template/mapper/device/device.go [hunk 5]
```diff
@@ -268,25 +270,35 @@ func setVisitor(visitorConfig *driver.VisitorConfig, twin *common.Twin, dev *dri
...
-	devs := make(map[string]*common.DeviceInstance)
-
-	switch cfg.DevInit.Mode {
-	//case common.DevInitModeConfigmap:
-	//	if err := parse.Parse(cfg.DevInit.Configmap, devs, d.models, d.protocols); err != nil {
-	//		return err
-	//	}
-	case common.DevInitModeRegister:
-		if err := parse.ParseByUsingRegister(devs, d.models, d.protocols); err != nil {
+func (d *DevPanel) DevInit(deviceList []*dmiapi.Device, deviceModelList []*dmiapi.DeviceModel) error {
```
Comment (WillardHu) [inline staging/src/github.com/kubeedge/mapper-framework/_template/mapper/device/device.go:273]: Consider adding some unit tests to your new method

File: staging/src/github.com/kubeedge/mapper-framework/_template/mapper/cmd/main.go [hunk 6]
```diff
@@ -27,29 +27,35 @@ func main() {
...
 		},
-		device.NewDevPanel(),
+		panel,
 	)
-
-	panel := device.NewDevPanel()
-	err = panel.DevInit(c)
-	if err != nil && !errors.Is(err, parse.ErrEmptyData) {
+	defer grpcServer.Stop()
+	if err = grpcServer.Start(); err != nil {
```
[the remark this answers is not in the document]
Comment (cl2017) [reply staging/src/github.com/kubeedge/mapper-framework/_template/mapper/cmd/main.go:54]: yes, there is just a change of order.

File: staging/src/github.com/kubeedge/mapper-framework/_template/mapper/cmd/main.go [hunk 7]
```diff
@@ -9,9 +9,9 @@ import (
 	"github.com/kubeedge/Template/device"
 	"github.com/kubeedge/Template/pkg/common"
 	"github.com/kubeedge/Template/pkg/config"
+	"github.com/kubeedge/Template/pkg/grpcclient"
 	"github.com/kubeedge/Template/pkg/grpcserver"
 	"github.com/kubeedge/Template/pkg/httpserver"
-	"github.com/kubeedge/Template/pkg/util/parse"
 )
 
 func main() {
```
[the remark this answers is not in the document]
Comment (cl2017) [reply staging/src/github.com/kubeedge/mapper-framework/_template/mapper/cmd/main.go:17]: ok, I will be completed in the new PR

File: staging/src/github.com/kubeedge/mapper-framework/_template/mapper/device/device.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (cl2017) [reply staging/src/github.com/kubeedge/mapper-framework/_template/mapper/device/device.go]: ok, done

File: staging/src/github.com/kubeedge/mapper-framework/_template/mapper/device/device.go (hunk 2 above)
Comment (cl2017) [reply staging/src/github.com/kubeedge/mapper-framework/_template/mapper/device/device.go]: ok, done

File: staging/src/github.com/kubeedge/mapper-framework/_template/mapper/cmd/main.go (hunk 6 above)
[the remark this answers is not in the document]
Comment (WillardHu) [reply staging/src/github.com/kubeedge/mapper-framework/_template/mapper/cmd/main.go:54]: I think this function `grpcServer.Start()` is a blocking function, originally it is at the end of the main function so it doesn't matter, but changing the order will cause later code is blocked.

File: staging/src/github.com/kubeedge/mapper-framework/_template/mapper/device/device.go [hunk 8]
```diff
@@ -29,6 +29,7 @@ type DevPanel struct {
 	deviceMuxs   map[string]context.CancelFunc
 	devices      map[string]*driver.CustomizedDev
 	models       map[string]common.DeviceModel
+	protocols    map[string]common.ProtocolConfig
```
Comment (wbc6080) [inline staging/src/github.com/kubeedge/mapper-framework/_template/mapper/device/device.go:32]: In #5305 we have removed the protocols field. It seems you made some mistakes while rebasing @cl2017

File: staging/src/github.com/kubeedge/mapper-framework/pkg/global/global.go [hunk 9]
```diff
@@ -1,18 +1,18 @@
...
 // DevPanel defined operations on devices, manage the lifecycle of devices
 type DevPanel interface {
 	// DevStart start device to collect/push/save data to edgecore/app/database
 	DevStart()
 	// DevInit get device info by dmi interface
-	DevInit(cfg *config.Config) error
+	DevInit(deviceList []*dmiapi.Device, deviceModelList []*dmiapi.DeviceModel) error
 	// UpdateDev update device's config and restart the device
-	UpdateDev(model *common.DeviceModel, device *common.DeviceInstance)
+	UpdateDev(model *common.DeviceModel, device *common.DeviceInstance, protocol *common.ProtocolConfig)
```
Comment (wbc6080) [inline staging/src/github.com/kubeedge/mapper-framework/pkg/global/global.go:15]: It seems function input does not correspond to the actual call involved
```suggestion
	UpdateDev(model *common.DeviceModel, device *common.DeviceInstance)
```

x3

**Review by wbc6080 — COMMENTED** [review]
/lgtm
cc @WillardHu @fisherxu

**Comment by WillardHu** [comment]
Please make the checks successful.

**Pull request merged**